src/Type/Php/ArrayMergeFunctionDynamicReturnTypeExtension.php

```diff
@@ -30,6 +32,41 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 			return ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getReturnType();
 		}
 
+		$allConstant = true;
+		foreach ($functionCall->getArgs() as $arg) {
+			$argType = $scope->getType($arg->value);
+			if ($arg->unpack || !$argType instanceof ConstantArrayType) {
+				$allConstant = false;
```

Comment:
I like the result of this PR but the implementation could be improved - you're iterating over `->getArgs()` twice and you're getting the `$arg->value` type twice. Instead, you could save the argument types to an array and the second foreach could iterate over that.